modules/mempool.nix

```diff
@@ -0,0 +1,331 @@
+            Path of the static frontend content root.
+          ";
+        };
+        nginxConfig = mkOption {
+          readOnly = true;
+          default = frontend.nginxConfig;
+          defaultText = "(See source)";
+          description = mdDoc "
+            An attrset of nginx config snippets for assembling a custom
+            mempool nginx config.
+            For details, see the source comments at the point of definition.
+          ";
+        };
+      };
+
+      address = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = mdDoc "Mempool backend address.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 8999;
+        description = mdDoc "Mempool backend port.";
+      };
+      electrumServer = mkOption {
+        type = types.enum [ "electrs" "fulcrum" ];
+        default = "electrs";
+        description = mdDoc ''
+          The Electrum server to use for fetching address information.
+
+          Possible options:
+          - electrs:
+            Small database size, slow when querying new addresses.
+          - fulcrum:
+            Large database size, quickly serves arbitrary address queries.
+        '';
+      };
+      settings = mkOption {
+        type = with types; attrsOf (attrsOf anything);
+        example = {
+          MEMPOOL = {
+            POLL_RATE_MS = 3000;
+            STDOUT_LOG_MIN_PRIORITY = "debug";
+          };
+          PRICE_DATA_SERVER = {
+            CLEARNET_URL = "https://myserver.org/prices";
+          };
+        };
+        description = mdDoc ''
+          Mempool backend settings.
+          See here for possible options:
+          https://github.com/mempool/mempool/blob/master/backend/src/config.ts
+        '';
+      };
+      database = {
+        name = mkOption {
+          type = types.str;
+          default = "mempool";
+          description = mdDoc "Database name.";
+        };
+      };
+      package = mkOption {
+        type = types.package;
+        default = nbPkgs.mempool-backend;
+        defaultText = "config.nix-bitcoin.pkgs.mempool-backend";
+        description = mdDoc "The package providing mempool binaries.";
+      };
+      user = mkOption {
+        type = types.str;
+        default = "mempool";
+        description = mdDoc "The user as which to run Mempool.";
+      };
+      group = mkOption {
+        type = types.str;
+        default = cfg.user;
+        description = mdDoc "The group as which to run Mempool.";
+      };
+      tor = nbLib.tor;
+    };
+
+    # Internal read-only options used by `./nodeinfo.nix` and `./presets/enable-tor.nix`.
+    mempool-frontend = let
+      copyValue = default: mkOption {
+        internal = true;
+        readOnly = true;
+        inherit default;
+      };
+    in {
+      enable = copyValue cfg.frontend.enable;
+      address = copyValue cfg.frontend.address;
+      port = copyValue cfg.frontend.port;
+    };
+  };
```

Comment:
I don't quite understand how this works. onion-services.nix sets mempool-frontend `externalPort` and the `port` is the default `mempool.frontend.port`. What if the user sets `mempool.frontend.port`? In that cas, doesn't that lead to the creating an onion service for the wrong port?

Reply:
No, `mempool.frontend.port` always equals `mempool-frontend.port`. I've renamed `copyValue` to `mkAlias`.